### DISCUSSION TITLE
where to set mount options in case of pam_mount usage

### DISCUSSION THREAD

**Discussion by thetuxinator** [rfjakob/gocryptfs#743] [Q&A]
Hi all, thx for this nice piece of software!

My only remaining Problem is, i am on btrfs and i want to add the noprealloc mount option, for performance reasons.
I am a very advanced Linux  user and i looked almost everywhere in /etc/
What i tried so far:

in the file /etc/security/pam_mount.conf.xml i added 
`<mntoptions allow="nosuid,nodev,loop,encryption,fsck,nonempty,allow_root,allow_other,noprealloc" />`

and 

```
<volume
  user="myuser"
  fstype="fuse"
  options="nodev,nosuid,quiet,allow_other,noprealloc"
  path="/usr/local/bin/gocryptfs#/home/%(USER).cipher"
  mountpoint="/home/%(USER)"
        />
```

any hint what i may have missed?

thx

Kind regards

**Comment by rfjakob** [comment]
Looking at `mntoptions allow` in https://man.archlinux.org/man/community/pam_mount/pam_mount.conf.5.en , this seems to only affect per-user configuration files. So in your case I think it does nothing.

The `volume` parts looks fine and matches what https://github.com/rfjakob/gocryptfs/wiki/Mounting-on-login-using-pam_mount#pam_mount-config says.

It does not work I guess? Do you get a mount error or is the option not applied?

**Comment by thetuxinator** [comment]
> Looking at `mntoptions allow` in https://man.archlinux.org/man/community/pam_mount/pam_mount.conf.5.en , this seems to only affect per-user configuration files. So in your case I think it does nothing.
> 
> The `volume` parts looks fine and matches what https://github.com/rfjakob/gocryptfs/wiki/Mounting-on-login-using-pam_mount#pam_mount-config says.
> 
> It does not work I guess? Do you get a mount error or is the option not applied?
Everything works and it gets mounted, it is just missing the noprealloc option ;-)
Any good idea on how to add that option? :D

**Comment by thetuxinator** [comment]
For info, i am trying to understand the pam_mount sourcecode and will update here if i find a solution

**Comment by lechner** [comment]
Hi @thetuxinator 

I am not sure I understand the `pam_mount` source code but I successfully used this file as a user config after the maintainer accepted [this commit](https://codeberg.org/jengelh/pam_mount/commit/2f50275bcb195466fadddc73b642a13a927af391). In XML parlance, I think you are looking for the `options` attribute.
```
<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE pam_mount SYSTEM "pam_mount.conf.xml.dtd">
<!--
	See pam_mount.conf(5) for a description.
-->

<pam_mount>

<volume
	path="/usr/bin/gocryptfs#/acct/lechner/home.wallace.gocryptfs"
	mountpoint="/home/lechner"
	user="lechner"
	fstype="fuse"
	options="nodev,nosuid"
/>

</pam_mount>
```
Kind regards
Felix Lechner

**Reply by thetuxinator** [reply]
When i add it in that options part, it seems to be ignored and "mount" does not show up the option.

kind regards
